previous_code/oracle.py

```diff
@@ -0,0 +1,511 @@
+'''import statements'''
```

Comment:
This is really hard to review. What has really changed with respect to the original `oracle.py`?